First round at supporting sending via Camel

pilhuhn opened the pull request:
This is a WIP, I don't know what I am doing here. And due to my node-sass issues, I can't even tell if I would see anything on screeen.
This uses the OpenAPI from https://github.com/RedHatInsights/notifications-backend/pull/237

pilhuhn committed:
First round at supporting sending via Camel

pilhuhn committed:
Rename to match schema.

josejulio commented on src/types/Integration.ts:
```diff
@@ -25,11 +27,20 @@ export interface IntegrationHttp extends IntegrationBase {
+    sslVerificationEnabled: boolean;
+    secretToken?: string;
+    basicAuth?: string; // TODO this is a map { user, pass}
+    extras?: string; // TODO this is a freeform map<string,string>
```
more or less

```suggestion
    basicAuth?: {
        user: string;
        pass: string;
    };
    extras?: Record<string, string>;
```

josejulio commented on src/generated/OpenapiIntegrations.ts:
```diff
@@ -31,32 +31,6 @@ export namespace Schemas {
     username?: string | undefined | null;
   };
 
-  export const BehaviorGroup = zodSchemaBehaviorGroup();
```
These are hidden from the openapi in current's `notification-backend`. I left them here because I'm using it for the behavior group work.

I'll probably move them to other file to allow the code to compile and play nice with the schema generator.

pilhuhn committed:
Update src/types/Integration.ts

Co-authored-by: Josejulio Martínez <finwemartinez@gmail.com>

pilhuhn committed:
Bring back behaviour groups.

pilhuhn committed:
Some fixes

josejulio committed:
Update schema

josejulio committed:
Refactor some weird types
 - UserIntegrationType is now a subset of IntegrationType instead of a
   new whole type
- UserIntegration is now a kind of Integration and is easier to udpate
  instead of having to add the specifics all over the place

josejulio commented:
Rebased, updated schema and improved (I hope) some types
